server/src/repositories/database.repository.ts

```diff
@@ -131,29 +113,34 @@ export class DatabaseRepository {
 
       const table = await this.getIndexTable(index);
       const dimSize = await this.getDimSize(table);
-      await this.dataSource.manager.transaction(async (manager) => {
-        await this.setSearchPath(manager);
-        await manager.query(`DROP INDEX IF EXISTS ${index}`);
-        await manager.query(`ALTER TABLE ${table} ALTER COLUMN embedding SET DATA TYPE real[]`);
-        await manager.query(`ALTER TABLE ${table} ALTER COLUMN embedding SET DATA TYPE vector(${dimSize})`);
-        await manager.query(`SET vectors.pgvector_compatibility=on`);
-        await manager.query(`
-          CREATE INDEX IF NOT EXISTS ${index} ON ${table}
+      await this.db.transaction().execute(async (tx) => {
+        await this.setSearchPath(tx);
+        await sql`DROP INDEX IF EXISTS ${sql.raw(index)}`.execute(tx);
+        await sql`ALTER TABLE ${sql.raw(table)} ALTER COLUMN embedding SET DATA TYPE real[]`.execute(tx);
+        await sql`ALTER TABLE ${sql.raw(table)} ALTER COLUMN embedding SET DATA TYPE vector(${sql.raw(String(dimSize))})`.execute(
+          tx,
+        );
+        await sql`SET vectors.pgvector_compatibility=on`.execute(tx);
+        await sql`
+          CREATE INDEX IF NOT EXISTS ${sql.raw(index)} ON ${sql.raw(table)}
           USING hnsw (embedding vector_cosine_ops)
-          WITH (ef_construction = 300, m = 16)`);
+          WITH (ef_construction = 300, m = 16)
+        `.execute(tx);
       });
     }
   }
 
+  @GenerateSql({ params: [VectorIndex.CLIP] })
   async shouldReindex(name: VectorIndex): Promise<boolean> {
     if (this.vectorExtension !== DatabaseExtension.VECTORS) {
       return false;
     }
 
     try {
-      const query = `SELECT idx_status FROM pg_vector_index_stat WHERE indexname = $1`;
-      const res = await this.dataSource.query(query, [name]);
-      return res[0]?.['idx_status'] === 'UPGRADE';
+      const { rows } = await sql<{
+        idx_status: string;
+      }>`SELECT idx_status FROM pg_vector_index_stat WHERE indexname = ${sql.val(name)}`.execute(this.db);
+      return rows[0]?.idx_status === 'UPGRADE';
     } catch (error) {
       const message: string = (error as any).message;
       if (message.includes('index is not existing')) {
```

Comment:
The error structure is different now, so this was always false and broke the upgrade process. Oh, JavaScript...

```suggestion
      const message: string = (error as any).error;
      if (message.includes('index is not existing')) {
```

Reply:
OK. I have no idea what it should be though. So I'll leave it as is and you can push appropriate changes if you'd like to.

Reply:
Oh, false alarm. I just needed to restart the server one more time. Hmm, it is weird that I needed to do that, though.

Reply:
ah, I see you included a suggestion.

Reply:
OH. It's just the failed query getting logged by kysely, not the actual error in the server. `message` was correct for the error object here after all.